Remove Jackson dep

bensigelman opened the pull request:
In the process, make Object->JSON work less-well... but not a big deal in light of https://github.com/lightstep/lightstep-tracer-java/issues/39

Also, add support for `Span.setOperationName` which was recently incorporated into the OT Span API.

bcronin commented:
This looks good. Removing this dependency had been on the TODO list for a _while_. Thanks!

bensigelman commented on common/src/main/java/com/lightstep/tracer/shared/Span.java:
```diff
@@ -124,31 +121,20 @@ public AbstractTracer getTracer() {
-          log.setPayload_json(payload.toString());
-        }
-      }
+      log.setPayload_json(JSONObject.valueToString(payload));
```
Great suggestion, thanks.

bensigelman commented:
BTW, I realized I was using the wrong / non-standard JSONObject artifact... adjusted that and also adjusted the code here accordingly. There is no longer an exception, btw, since we're using an overridden `toString()` which is of course a method on `j.l.Object`.
